cmd/gcp-controller-manager/app/node_annotater_test.go

```diff
@@ -182,7 +179,7 @@ func TestExtrackKubeLabels(t *testing.T) {
 			if got, want := out, c.out; !reflect.DeepEqual(got, want) {
 				t.Errorf("unexpected labels\n\tgot:\t%v\n\twant:\t%v", got, want)
 			}
-			if got, want := (err != nil), c.expectErr; got != want {
+			if got, want := (err != nil) && (err != errNoMetadata), c.expectErr; got != want {
```

Comment:
Check this explicitly in test cases. Either by setting c.expectErr or (better) by adding new c.expectNoMetadataErr

Reply:
Done.